data/db-router.js

```diff
@@ -0,0 +1,98 @@
+const router = require('express').Router();
+
+const db = require('./db.js');
+
+router.post('/', (req, res) => {
+  const newPost = req.body;
+  if (newPost.title && newPost.contents) {
+    db.insert(newPost)
+      .then(post => {
+        res.status(201).json(newPost);
+      })
+      .catch(err => {
+        res.status(500).json({
+          error: 'There was an error while saving the post to the database'
+        });
+      });
+  } else {
+    res.status(400).json({
+      errorMessage: 'Please provide title and contents for the post.'
+    });
+  }
+});
+
+router.get('/', (req, res) => {
+  db.find()
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The posts information could not be retrieved.' });
+    });
+});
+
+router.get('/:id', (req, res) => {
+  db.findById(req.params.id)
+    .then(post => {
+      if (post[0]) {
+        res.status(200).json(post[0]);
+      } else {
+        res
+          .status(404)
+          .json({ message: 'The post with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The post information could not be retrieved.' });
+    });
+});
+
+router.delete('/:id', (req, res) => {
+  db.findById(req.params.id)
+    .then(post => {
+      if (post[0]) {
+        db.remove(req.params.id).then(res.status(200).json(post[0]));
+      } else {
+        res
+          .status(404)
+          .json({ message: 'The post with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res.status(500).json({ error: 'The post could not be removed' });
+    });
+});
+
+router.put('/:id', (req, res) => {
+  const { title, contents } = req.body;
+
+  db.findById(req.params.id)
+    .then(post => {
+      if (post[0]) {
+        if (title && contents) {
+          db.update(req.params.id, req.body).then(post => {
+            res.status(200).json(req.body);
```

Comment:
just like with the successful POST request, you're only sending back what the user sent into the endpoint here -- you want to return the updated post as it exists within the database including its id and timestamp properties, so you need to perform another db call depending on a successful PUT request here